src/common_utils/language_utils/sentence_filter_utils.py

```diff
@@ -101,12 +117,36 @@ def filter_sentence(self, sentence, forms_to_filter):
                        sentence_after_extraction))
         return list(map(lambda x: x[0].lower(), sentence_filtered))
 
+    def filter_sentence_complex(self, sentence):
+        words = list(filter(lambda y: y.lower() not in self.stop_words, sentence.split(' ')))
+        sentence_after_extraction = list(map(lambda z: self.extract_lemma(z), words))
+        sentence_after_extraction = list(filter(lambda x_list: not is_empty_list(x_list), sentence_after_extraction))
+        sentence_filtered = list(map(lambda x_list: list_to_str_with_colons(x_list), sentence_after_extraction))
+        sentence_filtered = list(filter(lambda y: y.lower() not in self.stop_words, sentence_filtered))
+        sentence_filtered = list(map(lambda y: y.lower(), sentence_filtered))
```

Comment:
wouldn't it be better to map these words to `lower()` at the beginning? it wouldn't be necessary to call it twice then

Reply:
and, is it possible to refactor this method somehow? maybe these operations can be chained? it is not highly readable now